src/compute/datomic_client_memdb/core.clj

```diff
@@ -61,11 +70,17 @@
 
   client-impl/Queryable
   (q [_ arg-map]
-    (apply peer/q (:query arg-map) (map (fn [x]
-                                          (if (instance? LocalDb x)
-                                            (.-db x)
-                                            x))
-                                        (:args arg-map))))
+    (let [{:keys [query args]} arg-map]
+      (when (collection-query? query)
+        (throw (ex-info "Only find-rel elements are allowed in client find-spec, see http://docs.datomic.com/query.html#grammar"
+                        {:cognitect.anomalies/category :cognitect.anomalies/incorrect,
+                         :cognitect.anomalies/message  "Only find-rel elements are allowed in client find-spec, see http://docs.datomic.com/query.html#grammar"
+                         :dbs (filterv #(instance? LocalDb %) args)})))
```

Comment:
Actually this should probably be `(filterv #(satisfies? client-proto/Db %) args)`